lib/msal-common/src/client/UsernamePasswordClient.ts

```diff
@@ -90,7 +90,7 @@ export class UsernamePasswordClient extends BaseClient {
 
         parameterBuilder.addClientId(this.config.authOptions.clientId);
         parameterBuilder.addUsername(request.username);
-        parameterBuilder.addPassword(request.password);
+        parameterBuilder.addPassword(encodeURIComponent(request.password));
```

Comment:
URI encoding is usually done in `RequestParameterBuilder`, can we do it there instead? And please check if username needs it as well

Reply:
+1, this will be needed for the usename.

Reply:
I added it to username as well, moved the encoding to the RequestParamaterBuilder methods.